### ISSUE TITLE
Separate pre-publish and post-publish panels visibility and settings

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#57268]
### Description

In the publishing flow, there's something that has always puzzled me a little. As a user, when I publish a post what I get is that the user interface shows me something _before_ publishing and something else _after_ publishing.

- Before: it shows me a series of _checks_: visibility and publish date settings, plus some suggestions like reminders to add tags and categories. Those are all useful things to check and make perfectly sense.
- After: there aren't checks. They're more suggestions on what to do next or available actions like 'view post'.

As such, the nature of the pre and post publish panels is very different conceptually. 

The problem with this is that the more I publish posts, the more I don't understand why _after_ publishing I get a checkbox that says:

`Always show pre-publish checks.`

It refers to 'pre' publish checks, but I'm gettint this checkbox in the _post_ publish panel, which doesn't make much sense. As a user, I'd be pretty confused. What this checkbox exactly enables / disables?

More importantly: unchecking thie checkbox disableds _both_ the pre and post publish panels. It only refers to the 'pre' publish checks but it also disables the _post_ publish suggestions. To me, this ic clearly wrong and not a good user experience.

Additionally: both panels provide slots that can be filled by plugins to add checks and suggestions. Users may find useful to disabled only one of the two panels and not be forced to disable or enable both of them at once. Plugins also may find useful to provide different flows depending whterh one or both panels are enabled or disabled.

Overall, I'd think these two panels should be completely separated and have two different settings to toggel their visibility.

Worth also reminding that in the Preferences modal dialog there's only one checkbox that disables both although it only refers to the 'pre' publish checks.

Screenshots of the two panels, without and with additional content added by plugins:

![prepublish and postpublish](https://github.com/WordPress/gutenberg/assets/1682452/e68e3461-1d5b-4c77-9791-c0e672618bf7)

Screenshot of the related setting in the Preferences modal dialog, where also the description only refers to the 'pre' publish checks:

![pre-publish checks preference](https://github.com/WordPress/gutenberg/assets/1682452/2da9f55f-397b-4a02-aac2-bb7011cb9889)

### Step-by-step reproduction instructions

- Publish a post.
- Observe both the pre and the pos publish panels show a checkbox labeled `Always show pre-publish checks.`
- Either uncheck the checkbox or disable the related setting in the Preferences modal dialog.
- Publish another post.
- Observe that _both_ the pre and post publish panels are disabled.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
